platform/event/event_descriptor.proto

```diff
@@ -0,0 +1,77 @@
+ * Unique identifier for an event.
+ *
+ * Contains the hash of the event, the creator identifier, the birth round, and the generation.
+ *
+ * An event's descriptor is constructed individually by each node that receives a `GossipEvent`,
+ * to uniquely identify that event. An event's descriptor isn't part of the `GossipEvent` itself,
+ * since the descriptor contains the fields `hash` and `generation`, which can be computed locally.
+ * Nodes receiving a `GossipEvent` have the required information to construct the event descriptor
+ * immediately upon receiving the event, without needing to wait for the event to reach consensus.
+ *
+ * Aside from being a unique identifier for events that have been received through gossip,
+ * the event descriptor contains the necessary information to describe an event's parents,
+ * in the `parents` field of `GossipEvent`.
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.event;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.event";
+// <<<pbj.java_package = "com.hedera.hapi.platform.event">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * Unique identifier for an event.
+ */
+message EventDescriptor {
+  /**
+   * The hash of the event.<br/>
+   * <blockquote>Future work<br/><blockquote>
+   * Future work: the precise hash methodology hasn't been determined yet.</blockquote></blockquote>
+   */
+  bytes hash = 1;
+
+  /**
+   * A Creator Identifier.<br/>
+   * This SHALL be the unique identifier for the node that created the event.
+   */
+  int64 creator_id = 2;
+
+  /**
+   * The birth round of the event.<br/>
+   * The birth round SHALL be the pending consensus round at the time the event is created.<br/>
+   * The pending consensus round SHALL be **one greater** than the latest round to reach consensus.
+   */
+  int64 birth_round = 3;
+
+  /**
+  * The generation of the event.<br/>
+  * This value SHALL be **one greater** than the _maximum_ generation of all parents.<br/>
+  */
+  int64 generation = 17;
```

Comment:
17?

Reply:
See the comment from Jasper above https://github.com/hashgraph/hedera-protobufs/pull/338#discussion_r1550128991

I'm not sure if it makes sense to document this reason in the comment for the field since their style seems so... formal

Reply:
You could add a line-end comment.  Perhaps something like `// This field is temporary until birth_round is complete.`

Reply:
Done